### ISSUE TITLE
Form elements with non-unique id issue reported in browser console

### ISSUE THREAD

**Issue by SMerdzhanov** [mantinedev/mantine#1881]
### What package has an issue

@mantine/form

### Describe the bug

Hey, I've been experiencing a rather odd issue with `useForm` from the `@mantine/form` package.

Let me try to describe it...

## Description

I'm building a pretty **standard sign up form** with name (full name, one text input), email, password, checkbox for newsletter consent and submit button.

I'm experiencing the exact same behavior both with my code and any Mantine form example from the docs to make sure that I'm not missing something.

## Current behavior

Including a `PasswordInput` next to any other inputs in a form, without explicitly specifying the input element ID but rely on the auto-generated one, results in a non-unique element IDs issue reported by the browser.

**The issue is that anytime a `PasswordInput` is included to any form, the browser console reports an error about elements with non-unique ids.**

## Expected behavior

Auto-generated input IDs to be unique and not cause any non-unique ID errors in the browser console.

**Workaround:** Explicitly provide IDs to every input component in the form manually.

## Environment

TypeScript, React, NextJS, Mantine v4, Chrome Desktop

Desktop, OSX, Chrome Version 103.0.5060.134 (Official Build) (arm64), @mantine/core ^4.1.2, @mantine/form ^4.1.2, react ^17.0.2, next ^12.1.0

## Steps to reproduce

Implement a form that includes any fields and a `PasswordInput` component (required to reproduce).

**Here's an example form for testing purposes:**

```jsx
// sign-up-form.tsx
import { useForm } from "@mantine/form";
import { TextInput, Button, PasswordInput } from "@mantine/core";

export default function Demo() {
  const form = useForm({
    initialValues: { email: "", password: "" },

    validate: {
      password: (value) =>
        value.length < 6 ? "Password must have at least 6 letters" : null,
      email: (value) => (/^\S+@\S+$/.test(value) ? null : "Invalid email"),
    },
  });

  return (
    <form onSubmit={form.onSubmit(console.log)}>
      <TextInput
        mt="sm"
        label="Email"
        placeholder="Email"
        {...form.getInputProps("email")}
      />
      <PasswordInput
        mt="sm"
        label="Password"
        placeholder="Password"
        {...form.getInputProps("password")}
      />
      <Button type="submit" mt="sm">
        Submit
      </Button>
    </form>
  );
}
```

**Results in the following error:**

![image](https://user-images.githubusercontent.com/17785884/181121902-8d64ca06-3d94-44e8-9886-82ca88196d5c.png)

### In which browser did the problem occur

Chrome

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/dazzling-cannon-tozm41?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

### Possible fix

Frankly, I have no clue as to what causes the issue. I was thinking SSR vs CSR issue or potentially even hydration issue related to the NextJS nature but probably that's wrong.

I notice that this behavior is not present within CRA projects but solely in NextJS.

Please do note that the CodeSandbox has been implemented with CRA where the issue is not observed but could still serve as a reference.

Hope that I'm really just missing something.

**Comment by rtivital** [comment]
This issue is related to previous uuid generation on server – during SSR all auto-generated ids were an empty string, after hydration ids are set to the correct value. The issue was fixed in 5.0 for React 18. You can either upgrade to 5.0 or ignore the warning.

**Comment by SMerdzhanov** [comment]
Nice catch, thank you for the quick response and sharing the solution @rtivital! 👏
